File: src/main/java/fr/freebuild/claimremover/permissions/Permissions.java
package fr.freebuild.claimremover.permissions;

import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import fr.freebuild.claimremover.utils.PlayerUtils;

/**
 * Class that handle permissions of plugin
 */
public enum Permissions {
  CMD("claimremover.commands");

  private String perm;

  /**
   * Enum constructor
   * @param perm String permission
   */
  Permissions(String perm) {
    this.perm = perm;
  }

  /**
   * Define if the permission is set
   *
   * @param player Player on which check the permissions
   * @param permission Permission to check
   * @return Return a boolean to define if the permission is set
   */
  public static Boolean isSetOn(Player player, String permission) {
    return (player != null && player.hasPermission(permission));
  }

  /**
   * Define if the permission is set
   *
   * @param player Player on which check the permissions
   * @return Return a boolean to define if the permission is set
   */
  public Boolean isSetOn(Player player) {
    return (player != null && player.hasPermission(this.perm));
  }

  /**
   * Define if the permission is set on the CommandSender
   * If the CommandSender is not a Player (ex: console) the result is set with the value
   * of param defaultSender
   *
   * @param sender CommandSender on which check the permissions
   * @param defaultSender Value returned if sender is not Player
   * @return Return a boolean to define if the permission is set
   */
  public Boolean isSetOn(CommandSender sender, Boolean defaultSender) {
    if (sender instanceof Player)
      return this.isSetOn((Player) sender);
    return defaultSender;
  }

  /**
   * Define if the permission is set on the CommandSender
   * If the CommandSender is not a Player (ex: console) the result is set to true
   *
   * @param sender CommandSender on which check the permissions
   * @return Return a boolean to define if the permission is set
   */
  public Boolean isSetOn(CommandSender sender) {
    return this.isSetOn(sender, true);
  }

  /**
   * Define if the permission is set and print an error message
   *
   * @param player Player on which check the permissions
   * @return Return a boolean to define if the permission is set
   */
  public Boolean isSetOnWithMessage(Player player) {
    final Boolean isSet = this.isSetOn(player);

    if (!isSet) {
      Permissions.sendMessage(player);
    }

    return isSet;
  }

  /**
   * Send permission denied to the player
   *
   * @param player Player that must receive the message
   */
  public static final void sendMessage(Player player) {
    PlayerUtils.sendMessageConfig(player, "ErrorMessages.PermissionNotAllowed");
  }
}

File: src/main/scala/fr/freebuild/claimremover/Permissions.scala
package fr.freebuild.claimremover

import fr.freebuild.claimremover.ClaimRemoverPlugin.configs
import fr.freebuild.claimremover.utils.PlayerUtils
import org.bukkit.command.CommandSender

case class Permission(value: String) {
  def isSetOn(sender: CommandSender): Boolean = sender.hasPermission(value)

  def isSetOnWithMessage(sender: CommandSender): Boolean = {
    val isSet = isSetOn(sender)
    if (!isSet) PlayerUtils.sendMessage(sender, configs.language.errorMessages.permissionNotAllowed)
    isSet
  }
}

object Permissions {
  val CMD_ANALYZE = Permission("claimremover.commands.analyze")
  val CMD_CLEAR = Permission("claimremover.commands.clear")
  val CMD_CONFIG = Permission("claimremover.commands.config")
  val CMD_DELETE = Permission("claimremover.commands.delete")
  val CMD_EXPORT = Permission("claimremover.commands.export")
  val CMD_IMPORT = Permission("claimremover.commands.import")
  val CMD_INFO = Permission("claimremover.commands.info")
  val CMD_RELOAD = Permission("claimremover.commands.reload")
  val CMD_VERSION = Permission("claimremover.commands.version")
}
